virtcontainers/device/drivers/block.go

```diff
@@ -101,26 +110,19 @@ func (device *BlockDevice) Detach(devReceiver api.DeviceReceiver) error {
 		deviceLogger().WithError(err).Error("Failed to unplug block device")
 		return err
 	}
-	device.DeviceInfo.Hotplugged = false
+	device.AttachCount = 0
 	return nil
 }
 
-// IsAttached checks if the device is attached
-func (device *BlockDevice) IsAttached() bool {
-	return device.DeviceInfo.Hotplugged
-}
-
 // DeviceType is standard interface of api.Device, it returns device type
 func (device *BlockDevice) DeviceType() config.DeviceType {
 	return config.DeviceBlock
 }
 
-// DeviceID returns device ID
-func (device *BlockDevice) DeviceID() string {
-	return device.ID
-}
-
 // GetDeviceInfo returns device information used for creating
 func (device *BlockDevice) GetDeviceInfo() interface{} {
 	return device.BlockDrive
 }
+
+// It should implement GetAttachCount() and DeviceID() as api.Device implementation
+// here it shares function from *GenericDevice so we don't need duplicate codes
```

Comment:
What does this comment address to?

Reply:
Just tell the developers that as a `api.Device` implementation, it should define two function `GetAttachCount` and `DeviceID` but in this file there're no these function. Because it shares function s from embedded `*GenericDevice`